echo_api.rb

```diff
@@ -31,46 +32,68 @@ def get_headers
 end
 
 def echo_response
-  request.body.rewind
+  body_hasher = Digest::SHA1.new
+  body = ""
+
+  request.body.each do |chunk|
+    # Exclude obviously non-printable character(s).
+    body << chunk if /[\x00-\x1F]/ !~ chunk
```

Comment:
Only area I think is worth debating - if there's some unprintable character in a chunk I just skip it. This will cause the size to be misreported. Do we care? Easy to change, if so.